### PR TITLE
[SPARK-169] Re-implement !list & !quote's formatter by using a centralized template

### PR DESCRIPTION
**Pull request by theunkn0wn1**
This PR addresses [SPARK-169] by introducing `jinja2` for templating, and migrating both `!list` and `!quote`'s handlers to use it.

In basic, the core issue was the two different commands having a different bespoke renderers that do basically the same thing.
I previously took a stab at this in previous PRs, the most recent involving `__format__`. Ultimately im not pleased with that interface, so this is *yet another attempt* at it.

You may be thinking "why jinja" and "isn't jinja for HTML rendering?". To the former, its mainly due to it being a template engine, which makes it better suited to rendering the same stuff (templated stuff) with slight variations than pure python. To the latter, its not just:tm: limited to rendering HTML, it can be used for generating any textual documents. For another project i am using it for code generation (Something this project should hopefully never need....)

# Implementation details
The new render is split into three templates, `board`, `rescue`, `quotation`.
Three templates are used since each one has some special logic, and splitting it up keeps the (relative) complexity lower.

The templates are rendered asynchronously, since jinja supports that.

I ended up inserting a bunch of stuff into the template namespace to facilitate rendering, including functions to render the board & rescues. I also included some general purpose stuff, such as `pendulum.now` and the IRC text formatting functions `bold`, `italic`, `color` and the color enum `Color`.

Rendering of a rescue is primarially controlled by the passed `ListFlags` instance, which has been expanded somewhat to cover most the things the template needs to alter its output accordingly, as to minimize the string manipulation that needs to be done by hand.

Proposed output sample:
image

### CODE REVIEW

**Comment by Master-Guy** [comment]
[Error] Please rebase to develop before submitting the PR
[Testing] No unit tests?
[Consensus]/[Discussion] The output looks to be messy - Sometimes, but not always, empty lines after a rescue

```
[02-01-2021 10:14:30] <@Master-Guy> !inject MG CR 00:00 PC SOL
[02-01-2021 10:14:30] <@Master-Guy> !inject MG2 00:00 PC
[02-01-2021 10:14:30] <@Master-Guy> !inject MG3 PC
[02-01-2021 10:14:30] <@Master-Guy> !inject MG4 PS
[02-01-2021 10:14:30] <@Master-Guy> !assign 1 test_rat
[02-01-2021 10:14:30] <@Master-Guy> !active 2 client left

[02-01-2021 10:14:49] <@Master-Guy> !list -irc
[02-01-2021 10:14:49] <Mecha-Guy[BOT]> [0] MG's case (PC) in  <unspecified> 
[02-01-2021 10:14:49] <Mecha-Guy[BOT]>         No assigned Rats.  
[02-01-2021 10:14:49] <Mecha-Guy[BOT]> [3] MG4's case (PS) in  <unspecified> 
[02-01-2021 10:14:49] <Mecha-Guy[BOT]>         No assigned Rats.  
[02-01-2021 10:14:49] <Mecha-Guy[BOT]> [1] MG2's case (PC) in  <unspecified> 
[02-01-2021 10:14:49] <Mecha-Guy[BOT]>         Assigned Rats: 
[02-01-2021 10:14:49] <Mecha-Guy[BOT]>     Unidentified Rats: test_rat
[02-01-2021 10:14:49] <Mecha-Guy[BOT]>  
[02-01-2021 10:14:49] <Mecha-Guy[BOT]> [2] MG3's case (PC) in  <unspecified>  (Inactive) 
[02-01-2021 10:14:49] <Mecha-Guy[BOT]>         No assigned Rats.  
[02-01-2021 10:14:49] <Mecha-Guy[BOT]>  
```

**Comment by theunkn0wn1** [comment]
Updated against current develop.
Relevant tests are already implemented, no need to test precise output when the exact output may change down the line.
The existing tests ensure the relevant information is there.

Trailing newlines should be handled  now, mainly by calling `rstrip('\n')` after the renderer. less effort than figuring out where trailing white space is emitted from the templates.

File: src/commands/__init__.py [hunk 1]
```diff
@@ -22,4 +23,5 @@
     "starsystems",
     "administration",
     "api_utilities",
+    "RescueRenderFlags"
```
Comment (theunkn0wn1) [inline src/commands/__init__.py]: [error] this isn't part of command's namespace now.

File: src/commands/case_management.py [hunk 2]
```diff
@@ -40,6 +40,7 @@
 from ..packages.rat import Rat
 from ..packages.rescue import Rescue
 from ..packages.utils import Platforms, Status, color, bold, Colors
+from ..templates import env
```
Comment (theunkn0wn1) [inline src/commands/case_management.py]: [style] merge this with the import on line 23.

File: src/templates/render_flags.py [hunk 3]
```diff
@@ -41,4 +43,6 @@ def from_word(cls, argument: str):
         """
         show API UUIDs
         """
-        return cls(show_inactive, filter_unassigned_rescues, show_assigned_rats, show_uuids)
+        show_unident = True  # fixme remove hardcode
```
Comment (theunkn0wn1) [inline src/templates/render_flags.py]: FIXME

**Comment by theunkn0wn1** [comment]
Reviewing SPARK-169, both the inject handler and ~~grab~~ (not as of SPARK-199) should also use these new renderers.

**Comment by theunkn0wn1** [comment]
`!grab` doesn't require attention (thanks to SPARK-199), and `inject` is differed to SPARK-248.

**Comment by Master-Guy** [comment]
LGTM 👍 
Last style comment, but very non-blocking: The last rescue also gets an empty line at the end, which seems to be a little bit overkill. Removing this shouldn't affect a lot of things.

**Pull request merged**